src/db_loader.c

```diff
@@ -348,7 +358,27 @@ static void flush_symbol_buffer() {
 
 static void add_redis_pattern(char **composite_key, unsigned int arity, char *value) {
     char *key = expression_hash(EXPRESSION_HASH, composite_key, arity);
-    REDIS_APPEND_COMMAND_MACRO(REDIS, "SADD %s:%s %s", PATTERNS, key, value);
+
+    // Look up the key in the hash table
+    HashMap *entry;
+    HASH_FIND_STR(score_hash_map, key, entry);
+
+    // If key doesn't exist, initialize it with score = 0
+    unsigned int score = 0;
+    if (entry) {
+        score = entry->score;
+    } else {
+        entry = (HashMap *)malloc(sizeof(HashMap));
+        entry->key = strdup(key);
```

Comment:
Why do we need to `strdup()`? I think `expression_hash()` already return a dup'ed string so the pointer can be assigned directly.

Reply:
Right, done in [b4a9f92](https://github.com/singnet/das-metta-parser/pull/63/commits/b4a9f9255e83ed58e0b5a61e59550dfd742ce9b0)